File: mycollab-ui/src/main/scala/com/mycollab/eventmanager/EventBusFactory.scala
/**
 * This file is part of mycollab-ui.
 *
 * mycollab-ui is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-ui is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-ui.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.eventmanager

import com.google.common.eventbus.{EventBus, SubscriberExceptionContext, SubscriberExceptionHandler}
import com.mycollab.vaadin.ui.MyCollabSession
import com.mycollab.vaadin.ui.MyCollabSession._
import org.slf4j.LoggerFactory

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */

object EventBusFactory {
  
  private val LOG = LoggerFactory.getLogger(EventBusFactory.getClass)
  
  def getInstance(): EventBus = {
    var eventBus = MyCollabSession.getCurrentUIVariable(EVENT_BUS_VAL).asInstanceOf[EventBus]
    if (eventBus == null) {
      eventBus = new EventBus(new SubscriberEventBusExceptionHandler)
      MyCollabSession.putCurrentUIVariable(EVENT_BUS_VAL, eventBus)
    }
    eventBus
  }
  
  private class SubscriberEventBusExceptionHandler extends SubscriberExceptionHandler {
    def handleException(e: Throwable, context: SubscriberExceptionContext) {
      LOG.error("Error", e)
    }
  }
  
}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/EventBusFactory.kt
/**
 * mycollab-ui - Parent pom providing dependency and plugin management for applications
		built with Maven
 * Copyright © ${project.inceptionYear} MyCollab (support@mycollab.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.vaadin

import com.google.common.eventbus.EventBus
import com.google.common.eventbus.SubscriberExceptionContext
import com.google.common.eventbus.SubscriberExceptionHandler
import com.mycollab.vaadin.ui.MyCollabSession
import org.slf4j.LoggerFactory

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object EventBusFactory {
    private val LOG = LoggerFactory.getLogger(EventBusFactory::class.java)

    @JvmStatic fun getInstance(): EventBus {
        var eventBus = MyCollabSession.getCurrentUIVariable(MyCollabSession.EVENT_BUS_VAL) as EventBus?
        if (eventBus == null) {
            eventBus = EventBus(SubscriberEventBusExceptionHandler())
            MyCollabSession.putCurrentUIVariable(MyCollabSession.EVENT_BUS_VAL, eventBus)
        }
        return eventBus
    }

    private class SubscriberEventBusExceptionHandler : SubscriberExceptionHandler {
        override fun handleException(e: Throwable, context: SubscriberExceptionContext) {
            LOG.error("Error", e)
        }
    }
}